cross-chain/bob/contracts/BurnFromMintTokenPoolUpgradeable.sol

```diff
@@ -0,0 +1,163 @@
+}
+
+/// @notice Upgradeable BurnFromMintTokenPool that implements CCIP v1.6.0 interface
+/// @dev This is a working implementation for BOB deployment
+contract BurnFromMintTokenPoolUpgradeable is
+    IPoolV1,
+    ITypeAndVersion,
+    Initializable,
+    OwnableUpgradeable
+{
+    /// @notice The token this pool manages
+    IERC20 public s_token;
+
+    /// @notice Mapping of allowed addresses
+    mapping(address => bool) private s_allowList;
+
+    /// @notice The router address
+    address public s_router;
+
+    /// @notice The RMN proxy address
+    address public s_rmnProxy;
+
+    /// @notice The version of this contract
+    string public constant override typeAndVersion =
+        "BurnFromMintTokenPoolUpgradeable 1.6.0";
+
+    /// @notice Events
+    event Burned(address indexed sender, uint256 amount);
+    event Minted(
+        address indexed sender,
+        address indexed recipient,
+        uint256 amount
+    );
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @notice Burns tokens from the pool
+    function lockOrBurn(
+        Pool.LockOrBurnInV1 calldata lockOrBurnIn
+    ) external override returns (Pool.LockOrBurnOutV1 memory) {
+        require(msg.sender == s_router, "Only router");
+        emit Burned(msg.sender, lockOrBurnIn.amount);
+        // Burn tokens
+        IBurnMintERC20(address(s_token)).burnFrom(
+            msg.sender,
+            lockOrBurnIn.amount
+        );
+
+        return
+            Pool.LockOrBurnOutV1({
+                destTokenAddress: abi.encode(address(s_token)),
+                destPoolData: ""
+            });
+    }
+
+    /// @notice Mints tokens to the receiver
+    function releaseOrMint(
+        Pool.ReleaseOrMintInV1 calldata releaseOrMintIn
+    ) external override returns (Pool.ReleaseOrMintOutV1 memory) {
+        require(msg.sender == s_router, "Only router");
+        emit Minted(
+            msg.sender,
+            releaseOrMintIn.receiver,
+            releaseOrMintIn.amount
+        );
+        // Mint tokens
+        IBurnMintERC20(address(s_token)).mint(
+            releaseOrMintIn.receiver,
+            releaseOrMintIn.amount
+        );
+
+        return
+            Pool.ReleaseOrMintOutV1({
+                destinationAmount: releaseOrMintIn.amount
+            });
+    }
+
+    /// @notice Checks if a token is supported
+    function isSupportedToken(
+        address token
+    ) external view override returns (bool) {
+        return token == address(s_token);
+    }
+
+    /// @notice Get the token managed by this pool
+    function getToken() external view returns (address) {
+        return address(s_token);
+    }
+
+    /// @notice Checks if a chain is supported
+    function isSupportedChain(
+        uint64 /* remoteChainSelector */
+    ) external pure override returns (bool) {
+        return true; // For demo purposes, accept all chains
```

Comment:
Should we fix this before merging?

Reply:
added the single remote chain selector on the initialize() and tests. should be addresses fully now.